go/toolchains.rst

```diff
@@ -76,6 +76,7 @@ the flag ``--@io_bazel_rules_go//go/toolchain:sdk_version="version"`` where
 ``"version"`` is the SDK version you would like to build with, eg. ``"1.18.3"``.
 The SDK version can omit the patch, or include a prerelease part, eg. ``"1"``,
 ``"1.18"``, ``"1.18.0"``, and ``"1.19.0beta1"`` are all valid values for ``sdk_version``.
+You can also refer to a Go SDK by name of the target used to declare it.
```

Comment:
This is a smart trick. :)
There are two problem with it though which makes me think we should solve this differently:
- With Bzlmod, canonical repository names as returned by `ctx.attr.name` will become an implementation detail, they shouldn't (and can't easily) be passed as command-line arguments.
- This essentially shortcuts ordinary toolchain resolution to "just use this exact toolchain", which I think is better solved using `--extra_toolchains`.

How about adding a `config_setting` that matches by SDK type (`host`/`download` or perhaps `local`/`remote` to match the Java toolchains)?

Reply:
I've added `host` to be consistent with `go_register_toolchains(version = "host")`.